x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.test.ts

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { omit } from 'lodash/fp';
+
+import { getSimpleRuleAsMultipartContent } from '../../../../../../../../test/detection_engine_api_integration/security_and_spaces/tests/utils';
+import { ImportSuccessError } from '../utils';
+import {
+  getImportRulesRequest,
+  getImportRulesRequestOverwriteTrue,
+  getFindResult,
+  getResult,
+  getEmptyIndex,
+  getFindResultWithSingleHit,
+  getNonEmptyIndex,
+} from '../__mocks__/request_responses';
+import { createMockServer, createMockConfig, clientsServiceMock } from '../__mocks__';
+import { importRulesRoute } from './import_rules_route';
+
+describe('import_rules_route', () => {
+  let server = createMockServer();
+  let config = createMockConfig();
+  let getClients = clientsServiceMock.createGetScoped();
+  let clients = clientsServiceMock.createClients();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    server = createMockServer();
+    config = createMockConfig();
```

Comment:
Right here you will want to define a few default values for the config mock which will help with test results:

```ts
    config = createMockConfig();
    config = () => ({
      get: jest.fn(value => {
        switch (value) {
          case 'savedObjects.maxImportPayloadBytes': {
            return 10000;
          }
          case 'savedObjects.maxImportExportSize': {
            return 10000;
          }
          case 'xpack.siem.signalsIndex': {
            return DEFAULT_SIGNALS_INDEX;
          }
          default: {
            const dummyMock = jest.fn();
            return dummyMock();
          }
        }
      }),
      has: jest.fn(),
    });
```

That will give you decent output of keys so that later you won't have icky tests with `undefined-undefined` and they will read much smoother for future generations of programmers and Kibana hobbyists.

Reply:
That makes a lot more sense, and does indeed make the tests a lot nicer. Thanks!